liubov.tarasenko/homework2/task2_2/parenthesis.rb

```diff
@@ -0,0 +1,37 @@
+class String 
+
+  def valid?
+    puts
+    puts self
+    opened_arr = []    
+    self.each_char{ |char|
+      if [ '{', '[', '(', '<' ].any?{|x| x == char}     
+        opened_arr << char
+      elsif [ '}', ']', ')', '>' ].any?{|x| x == char} 
+        pair = find_pair(char)
+        if opened_arr.last == pair 
+          opened_arr.pop
+        else
+          return false
+        end
+      end
+    }
+    opened_arr.empty?
+  end
+
+  def find_pair(char)
+    return '{' if char == "}"
+    return '<' if char == ">"
+    return '[' if char == "]"
+    return '(' if char == ")"
+  end
+
+end 
+puts '<<}'.valid?
+puts '<<}>'.valid?
+puts '<<}>>'.valid?
+puts '<<{}>>'.valid? 
+puts '<([{})]>>'.valid?
+puts '<<{([])}>>'.valid?
+puts ')('.valid?
+puts '({}){}'.valid?
```

Comment:
Next time please make sure u have a newline at end of file

https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

Reply:
@lubo4kaTarasenko